addons/mail/static/src/core/common/store_service.js

```diff
@@ -260,7 +284,8 @@ export const storeService = {
                 if (![Record.one()[0], Record.many()[0]].includes(val?.[0])) {
                     continue;
                 }
-                Model.__rels__.set(name, { targetModel: val[1] });
+                const [, { targetModel, onDelete } = {}] = val;
+                Model.__rels__.set(name, { targetModel, onDelete });
```

Comment:
set complete `val[1]` directly instead of deconstruct to avoid diff every time we'll want to add something?